### PR TITLE
map design doc

### PR DESCRIPTION
**Pull request by jreadey**
Deisgn docs for Map support

### CODE REVIEW

**Review by mattjala — APPROVED** [review]
We'll discuss the design later, but the document can go in

File: docs/design/map/h5_map.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+This request returns value for given key
+
+Request GET /maps/&lt;map_id&gt;/key/&lt;key&gt;
+
+Request Elements:
+  None
+
+Returns:
+  Value for gieven key as JSON or binary blob.  Or 404 if not found
```
Comment (hyoklee) [inline docs/design/map/h5_map.md:87]: `gieven` -> `given`

File: docs/design/map/h5_map.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+This request returns information on a map object
+
+Request: GET /maps/&lt;map_id&gt;
+
+Request Elements:
+
+* verbose: provide additional information in the respose
+
+Returns: JSON describing map object.  If verbose is set, additional daa will be returned (e.g. number oF KV pairs)
```
Comment (hyoklee) [inline docs/design/map/h5_map.md:55]: `oF` -> `of`

File: docs/design/map/h5_map.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+Marker If provided, a key indicating that only keys that occur after the marker key will be returned.
+
+Returns: JSON of key value pairs.  Up to an internal limit.
+
+TBD: support binary responses
+
+### Delete a key
+
+This request removes a given key an its associatted value
```
Comment (hyoklee) [inline docs/design/map/h5_map.md:118]: `an its` -> `and its`
`associatted` -> `associated`

**Review by hyoklee — COMMENTED** [review]
@mattjala & @jreadey , please fix typos

**Comment by jreadey** [comment]
fixed typos!

**Pull request merged**